### PR TITLE
fix(executors): Support explicit generics in lambda handler input & add debug logging around epsagon input handling

### PR DESCRIPTION
**Pull request by amitlicht**

### CODE REVIEW

File: epsagon/src/main/java/com/epsagon/executors/BasePOJOExecutor.java [hunk 1]
```diff
@@ -53,15 +57,20 @@ public BasePOJOExecutor(Class<?> userHandlerClass) throws ExecutorException {
      * @return The input as an object
      */
     protected Object parseInput(InputStream input) throws IOException {
-        Type inputType = _userHandlerMethod.getParameterTypes()[0];
+        Type inputType = _userHandlerMethod.getGenericParameterTypes()[0];
```
Comment (amitlicht) [inline epsagon/src/main/java/com/epsagon/executors/BasePOJOExecutor.java:60]: Attn: only actual change here that's not extra debug logging.

Comment (galbash) [reply epsagon/src/main/java/com/epsagon/executors/BasePOJOExecutor.java:60]: looks good, does this solve all the use cases or just the manual overriding?

Comment (amitlicht) [reply epsagon/src/main/java/com/epsagon/executors/BasePOJOExecutor.java:60]: it solves all usecases when the input type is an actual generic type, e.g. `ServiceLambdaMapping<StartBatchRequest>`. 
But not when the input is a generic variable such as `ServiceLambdaMapping<R>`. 
In the NWEA the lambda handler is implemented in an abstract generic baseclass parameterized by R, and then the subclass declares an actual type for R, which is StartBatchRequest. However the handler's signature remains parameterized by R - so this is not solved by this fix, and they need to redefine the handler in the baseclass (with the actual type) as a workaround. 
Still this fix captures many other usecases and so it's worth inserting.

**Review by galbash — APPROVED** [review]

**Pull request merged**